libs/core/src/components/grid/grid.ts

```diff
@@ -0,0 +1,126 @@
+  columns?: string
+
+  @property({ attribute: 'gap', type: String })
+  gap?: GridGap
+
+  @property({ attribute: 'padding', type: String })
+  padding?: GridPadding
+
+  @property({ attribute: 'fluid', type: String })
+  fluid?: boolean
+
+  updated(changedProperties: Map<string | number | symbol, unknown>) {
+    super.updated(changedProperties)
+
+    let cssVariables = ''
+
+    if (changedProperties.has('columns')) {
+      const [columnsDesktop, columnsTablet, columnsMobile] =
+        this.columns?.split(' ').map(Number) || []
+
+      const columnProperties = [
+        { name: 'gds-columns-mobile', value: columnsMobile },
+        { name: 'gds-columns-tablet', value: columnsTablet },
+        { name: 'gds-columns-desktop', value: columnsDesktop },
+      ]
+
+      cssVariables += columnProperties
+        .filter(({ value }) => value !== undefined)
+        .map(({ name, value }) => `--${name}: ${value};`)
+        .join(' ')
+    }
+
+    if (changedProperties.has('gap')) {
+      const [gapDesktop, gapTablet, gapMobile] = this.gap?.split(' ') || []
+
+      const gapProperties = [
+        { name: 'gds-gap-desktop', value: gapDesktop },
+        { name: 'gds-gap-tablet', value: gapTablet },
+        { name: 'gds-gap-mobile', value: gapMobile },
+      ]
+
+      cssVariables += gapProperties
+        .filter(({ value }) => value !== undefined)
+        .map(
+          ({ name, value }) => `--${name}: var(--gds-sys-grid-gap-${value});`
+        )
+        .join(' ')
+    }
+
+    if (changedProperties.has('padding')) {
+      const [paddingDesktop, paddingTablet, paddingMobile] =
+        this.padding?.split(' ') || []
+
+      const paddingProperties = [
+        { name: 'gds-padding-desktop', value: paddingDesktop },
+        { name: 'gds-padding-tablet', value: paddingTablet },
+        { name: 'gds-padding-mobile', value: paddingMobile },
+      ]
+
+      cssVariables += paddingProperties
+        .filter(({ value }) => value !== undefined)
+        .map(
+          ({ name, value }) =>
+            `--${name}: var(--gds-sys-grid-padding-${value});`
+        )
+        .join(' ')
+    }
+
+    if (changedProperties.has('fluid') && this.fluid) {
+      cssVariables += `
+        --gds-grid-col-count: var(--gds-c);
+        --gds-grid-col-width: 200px;
+        --gap-count: calc(var(--gds-grid-col-count) - 1);
+        --total-gap-width: calc(var(--gap-count) * var(--gds-gap-column));
+        --gds-grid-col-width-max: calc(
+          (100% - var(--total-gap-width)) / var(--gds-grid-col-count)
+        );
+  
+        grid-template-columns: repeat(
+          auto-fill,
+          minmax(max(var(--gds-grid-col-width), var(--gds-grid-col-width-max)), 1fr)
+        );
+        text-wrap: balance;
+      `
+    }
+
+    const sheet = new CSSStyleSheet()
+    sheet.replaceSync(`:host {${cssVariables}}`)
+
+    if (this.shadowRoot) {
+      this.shadowRoot.adoptedStyleSheets = [
+        ...this.shadowRoot.adoptedStyleSheets,
```

Comment:
I suspect that this will cause a new set of css variables to be added each time a property is changed.

Since you create a new stylesheet on lines 112..113 and then add it together with existing stylesheets here, it will keep adding new copies on each update (though I haven't verified this in practice)

A better approach may be to have a state variable with the set of grid variables, then update that state variable when one of those properties changes, and then add a watcher on the state variable to update the `adoptedStyleSheets` by combining with the static `styles` from the class.

Something along these lines:

```ts
// State variable to keep track of the grid variables
@state()
private _gridVariables = {
  column: css`...`,
  gap: css`...`,
  ...etc
}

// Watch the grid properties and update state variable when something changes
@watch('columns')
private _handleColumnsChanged() {
  this._gridVariables = {
    ...this._gridVariables,
    column: css`...new column variabels...`
  }
}
// ...etc for the rest of the properties

// Watch the state variable and update `adoptedStyleSheets` when it changes
@watch('_gridVariables')
private _updateGridCss() {
  const sheet = new CSSStyleSheet()
  sheet.replaceSync(`:host {${Object.values(this._gridVariables).join('')}}`) // join the state object to a string
  
  if (this.shadowRoot) {
    this.shadowRoot.adoptedStyleSheets = [
      ...GdsGrid.styles, // This should get the static styles added to the class, rather than the accumulated set of stylesheets
      sheet
  }
}
```

Reply:
Updating the stylesheet not sure if this is the best way to do it but please check the update and let me know 🤘
It could potentially do with some further refactoring reducing repetition for updating props, suggestions are welcomed ✌️